### PR TITLE
MGMT-6438 Removed deprecated requirements env variables.

### PR DESCRIPTION
**Pull request by jakub-dzon**
Following environments variables support have been removed:
 - HW_VALIDATOR_MIN_CPU_CORES_WORKER
 - HW_VALIDATOR_MIN_CPU_CORES_MASTER
 - HW_VALIDATOR_MIN_RAM_GIB_WORKER
 - HW_VALIDATOR_MIN_RAM_GIB_MASTER
 - HW_VALIDATOR_MIN_DISK_SIZE_GIB
 - HW_INSTALLATION_DISK_SPEED_THRESHOLD_MS

The requirements customization can be made through `HW_VALIDATOR_REQUIREMENTS`.

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

**Comment by ronniel1** [comment]
I think that these env variables are used in test-infra and E2E tests.
Did you modify them first?

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -2891,7 +2891,17 @@ func (b *bareMetalInventory) processDiskSpeedCheckResponse(ctx context.Context,
 	if exitCode == 0 {
 		b.metricApi.DiskSyncDuration(h.ClusterID, *h.ID, diskPerfCheckResponse.Path, diskPerfCheckResponse.IoSyncDuration)
 
-		if diskPerfCheckResponse.IoSyncDuration > b.hwValidator.GetInstallationDiskSpeedThresholdMs() {
+		cluster, err := common.GetClusterFromDB(b.db, h.ClusterID, common.UseEagerLoading)
```
Comment (RazRegev) [inline internal/bminventory/inventory.go:2894]: NIT: maybe can be extracted to a separate func?

**Review by RazRegev — COMMENTED** [review]
looks good to me, as @ronniel1 wrote need to make sure it doesn't break the e2e tests

**Comment by jakub-dzon** [comment]
@ronniel1 - I will double check that I haven't missed anything. Putting on hold until it's confirmed.

**Comment by jakub-dzon** [comment]
assisted-test-infra script had to be updated in one more place: https://github.com/openshift/assisted-test-infra/pull/795

File: hack/generate.sh [hunk 2]
```diff
@@ -76,10 +76,6 @@ function generate_configuration() {
     docker run --rm -v ${__root}/config/onprem-iso-fcc.yaml:/config.fcc:z quay.io/coreos/fcct:release --pretty --strict /config.fcc > ${__root}/config/onprem-iso-config.ign
 }
 
-function generate_ocp_version() {
-  generate_configuration
-}
```
Comment (carbonin) [inline hack/generate.sh:81]: Why was this removed?

**Comment by carbonin** [comment]
@jakub-dzon if someone previously wasn't supplying `HW_VALIDATOR_REQUIREMENTS` would all hosts now fail validations?
Since defaults were set in code we didn't previously need to supply these env vars. I'm wondering if we should have kept the old defaults as the "default" version key.

File: hack/generate.sh (hunk 2 above)
Comment (jakub-dzon) [reply hack/generate.sh:81]: That was a temporary backwards-compatibility wrapper; it's no longer needed, as https://github.com/openshift/assisted-test-infra/commit/5ba6a083ee61e02afa70a30fd154be35700a82aa and https://github.com/openshift-assisted/assisted-installer-deployment/commit/9d34ed4518666b31784d6b85ece0838c71a0b341 were merged.

**Comment by carbonin** [comment]
Specifically after this is merged we're seeing logs like:
`time="2021-05-13T21:02:39Z" level=error msg="failed to refresh state of host e4745a34-d11a-41fc-a55d-cb0750753d3d cluster 8c430fca-e58e-45ec-b285-1562a56499b1" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).refreshClusterHosts" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:1527" error="requirements for version 4.8.0 not found" go-id=667 pkg=Inventory request_id=`

File: hack/generate.sh (hunk 2 above)
Comment (carbonin) [reply hack/generate.sh:81]: Okay, but to be clear this doesn't have anything to do with the rest of the changes made in this PR, right?
I'm asking because it doesn't seem related to the env var change.

Comment (jakub-dzon) [reply hack/generate.sh:81]: It's not strictly related. It's a clean-up after some other env variables-related PR.
